### PR TITLE
Replace deprecated Guava method createTempDir

### LINKED ISSUE #6908: Remove usages of deprecated Files.createTempDir
`com.google.common.io.Files#createTempDir` is deprecated.
`java.nio.file.Files#createTempDirectory` should be used instead.

- [ ] core module #7818
- [ ] plugin module
- [ ] testing module
- [ ] .mvn/modernizer/violations.xml

### PR DESCRIPTION
**Pull request by joshthoward**
Fixes #6908.

Replace `com.google.common.io.Files.createTempDir` with `java.nio.file.Files.createTempDirectory`. The return value of the replacement methods is `java.nio.Path` instead of `java.io.File`. I've made minimal changes to account for this difference.

### CODE REVIEW

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java [hunk 1]
```diff
@@ -28,38 +29,38 @@
...
 
     @Test
     public void testBackupStore()
             throws Exception
     {
         // backup first file
-        File file1 = new File(temporary, "file1");
-        Files.write("hello world", file1, UTF_8);
+        File file1 = temporary.resolve("file1").toFile();
+        Files.asCharSink(file1, UTF_8).write("hello world");
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java:41]: Why did this change?

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java [hunk 2]
```diff
@@ -28,38 +29,38 @@
...
 
         assertFalse(store.shardExists(uuid1));
         store.backupShard(uuid1, file1);
         assertTrue(store.shardExists(uuid1));
 
         // backup second file
-        File file2 = new File(temporary, "file2");
-        Files.write("bye bye", file2, UTF_8);
+        File file2 = temporary.resolve("file2").toFile();
+        Files.asCharSink(file2, UTF_8).write("bye bye");
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java:50]: Same here

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/TestFileBackupStore.java [hunk 3]
```diff
@@ -18,37 +18,39 @@
...
-        deleteRecursively(temporary.toPath(), ALLOW_INSECURE);
+        deleteRecursively(temporary, ALLOW_INSECURE);
     }
 
     @Test
     public void testFilePaths()
     {
         UUID uuid = UUID.fromString("701e1a79-74f7-4f56-b438-b41e8e7d019d");
-        File expected = new File(temporary, format("backup/70/1e/%s.orc", uuid));
+        File expected = temporary.resolve("backup").resolve("70").resolve("1e").resolve(format("%s.orc", uuid)).toFile();
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/TestFileBackupStore.java:53]: Format as follows for clarity:
```suggestion
        File expected = temporary.resolve("backup")
                .resolve("70")
                .resolve("1e")
                .resolve(format("%s.orc", uuid))
                .toFile();
```

However, there's no need to chained calls. You can do:

```java
File expected = temporary.resolve(format("backup/70/1e/%s.orc", uuid));
```

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestFileStorageService.java [hunk 4]
```diff
@@ -67,9 +70,9 @@ public void testGetFileSystemPath()
     public void testFilePaths()
     {
         UUID uuid = UUID.fromString("701e1a79-74f7-4f56-b438-b41e8e7d019d");
-        File staging = new File(temporary, format("staging/%s.orc", uuid));
-        File storage = new File(temporary, format("storage/70/1e/%s.orc", uuid));
-        File quarantine = new File(temporary, format("quarantine/%s.orc", uuid));
+        File staging = temporary.resolve("staging").resolve(format("%s.orc", uuid)).toFile();
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestFileStorageService.java:73]: No need for chained calls to resolve. See comment above.

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java [hunk 5]
```diff
@@ -367,7 +369,7 @@ public void testRewriter()
     public void testWriterRollback()
     {
         // verify staging directory is empty
-        File staging = new File(new File(temporary, "data"), "staging");
+        File staging = temporary.resolve("data").resolve("staging").toFile();
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java:372]: No need for chained calls to `resolve`

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java [hunk 6]
```diff
@@ -628,7 +630,7 @@ public static RaptorStorageManager createRaptorStorageManager(
     private static void assertFileEquals(File actual, File expected)
             throws IOException
     {
-        assertEquals(hash(actual, md5()), hash(expected, md5()));
+        assertEquals(Files.asByteSource(actual).hash(md5()), Files.asByteSource(expected).hash(md5()));
```
Comment (martint) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java:633]: Unrelated change?

**Review by martint — CHANGES_REQUESTED** [review]

File: plugin/trino-atop/src/test/java/io/trino/plugin/atop/TestAtopHang.java [hunk 7]
```diff
@@ -39,8 +39,8 @@
     public void setUp()
             throws Exception
     {
-        File tempPath = createTempDir();
-        copyExecutable("hanging_atop.sh", tempPath);
+        Path tempPath = createTempDirectory(null);
```
Comment (findinpath) [inline plugin/trino-atop/src/test/java/io/trino/plugin/atop/TestAtopHang.java:42]: Please do add (in a separate commit)

```
        deleteRecursively(tempPath, ALLOW_INSECURE);
```

in the `tearDown` method in order to leave the testing environment in a clean state.

Comment (findinpath) [reply plugin/trino-atop/src/test/java/io/trino/plugin/atop/TestAtopHang.java:42]: I happened to find another undeleted temp directory in: 

- `io.trino.plugin.hive.s3.TestHiveS3Config#testExplicitPropertyMappings` 
- `io.trino.plugin.hive.rubix.TestRubixCaching#initializeRubix(io.trino.plugin.hive.rubix.RubixConfig, java.util.List<io.trino.spi.Node>)`
- `io.trino.plugin.hive.parquet.TestParquetDecimalScaling#getBasePath` 

These usecases are unrelated to this PR. Maybe another small PR would be better fit to host them.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 8]
```diff
@@ -103,18 +103,17 @@ private static String prepareCassandraYaml()
     {
         String original = Resources.toString(getResource("cu-cassandra.yaml"), UTF_8);
 
-        File tempDirFile = createTempDir();
-        tempDirFile.deleteOnExit();
-        Path tmpDirPath = tempDirFile.toPath();
+        Path tmpDirPath = createTempDirectory(null);
```
Comment (findinpath) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:106]: Pls add logic for deleting the temporary directory in the `#close()` method.

You can use `io.trino.plugin.elasticsearch.ElasticsearchServer` as template for this change.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveCreateExternalTable.java [hunk 9]
```diff
@@ -67,22 +67,23 @@ public void testCreateExternalTableWithData()
 
         MaterializedResult result = computeActual("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '/') FROM test_create_external");
         String tablePath = (String) result.getOnlyValue();
-        assertThat(tablePath).startsWith(tableLocation.toURI().toString());
+        assertThat(tablePath).isEqualTo("file:" + tableLocation + "/");
```
[the remark this answers is not in the document]
Comment (findinpath) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveCreateExternalTable.java:70]: build the table location by hand: `file:/var/folders/1q/y42hmc4s3yl38kp0t0q142_c0000gn/T/11675979297384514711/data/` 

vs
use `tableLocation.toUri`: `file:///var/folders/1q/y42hmc4s3yl38kp0t0q142_c0000gn/T/11675979297384514711/data/`

[the remark this answers is not in the document]
Comment (joshthoward) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveCreateExternalTable.java:70]: @martint Basically what @findinpath said. The `toUri` has a `file:///` prefix whereas `tablePath` has a `file:/` prefix. I thought this was the simplest way to work around that.

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java (hunk 1 above)
Comment (joshthoward) [reply plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java:41]: I assume you mean replacing `Files.write` with `Files.asCharSink`? If so, `Files.write` is another deprecated method in Guava that seemed close enough.

Comment (martint) [reply plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/backup/AbstractTestBackupStore.java:41]: Ok, fair enough, but put it in a separate commit, since it’s not related to the createTempDir change

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java (hunk 8 above)
Comment (joshthoward) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:106]: This should be covered by `yamlFile.deleteOnExit();`

File: plugin/trino-atop/src/test/java/io/trino/plugin/atop/TestAtopHang.java (hunk 7 above)
Comment (joshthoward) [reply plugin/trino-atop/src/test/java/io/trino/plugin/atop/TestAtopHang.java:42]: Let me raise this in another PR.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveCreateExternalTable.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (findinpath) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveCreateExternalTable.java:70]:
```suggestion
        assertThat(tablePath).isEqualTo(tableLocation.toFile().toURI().toString());
```

It's the same result without doing magic code concatenation

[the remark this answers is not in the document]
Comment (joshthoward) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveCreateExternalTable.java:70]: Okay. That's a good suggestion. I'll update.
